x-pack/plugins/serverless_search/public/application/components/connectors/edit_service_type.tsx

```diff
@@ -62,24 +64,125 @@ export const EditServiceType: React.FC<EditServiceTypeProps> = ({ connector, isD
     },
   });
 
+  const getInitialOptions = () => {
+    return allConnectors.map((conn, key) => {
+      const _append: JSX.Element[] = [];
+      if (conn.isTechPreview) {
+        _append.push(
+          <EuiBadge key={key + '-preview'} iconType="beaker" color="hollow">
+            {i18n.translate(
+              'xpack.serverlessSearch.connectors.chooseConnectorSelectable.thechPreviewBadgeLabel',
+              { defaultMessage: 'Tech preview' }
+            )}
+          </EuiBadge>
+        );
+      }
+      if (conn.isBeta) {
+        _append.push(
+          <EuiBadge key={key + '-beta'} iconType={'beta'} color="hollow">
+            {i18n.translate(
+              'xpack.serverlessSearch.connectors.chooseConnectorSelectable.BetaBadgeLabel',
+              {
+                defaultMessage: 'Beta',
+              }
+            )}
+          </EuiBadge>
+        );
+      }
+      return {
+        _append,
+        _prepend: <EuiIcon size="l" type={conn.iconPath} />,
+        key: key.toString(),
+        label: conn.name,
+        serviceType: conn.serviceType,
+      };
+    });
+  };
+
+  const initialOptions = getInitialOptions();
+  const { euiTheme } = useEuiTheme();
+  const [selectedOption, setSelectedOption] = useState<Array<EuiComboBoxOptionOption<OptionData>>>(
+    []
+  );
+  const [selectedConnector, setSelectedConnector] = useState<Connector | null>(null);
+  const renderOption = (
+    option: EuiComboBoxOptionOption<OptionData>,
+    searchValue: string,
+    contentClassName: string
+  ) => {
+    const { _append, key, label, _prepend, serviceType } =
+      option as EuiComboBoxOptionOption<OptionData> & {
```

Comment:
:x: fix the type instead of casting

```typescript
type ExpandedComboBoxOption = EuiComboBoxOptionOption<OptionData> & {
  _append: JSX.Element[];
  _prepend: JSX.Element;
  serviceType: string;
};

const renderOption = (
    option: ExpandedComboBoxOption,
    searchValue: string,
    contentClassName: string
  ) => {
```

Or you can just add `_append, _prepend, serviceType` to `OptionData`

Reply:
If I fix the type. I get this error:

```
No overload matches this call.
  Overload 1 of 2, '(props: _EuiComboBoxProps<OptionData>): EuiComboBox<OptionData>', gave the following error.
    Type '(option: ExpandedComboBoxOption, searchValue: string, contentClassName: string) => React.JSX.Element' is not assignable to type '(option: EuiComboBoxOptionOption<OptionData>, searchValue: string, OPTION_CONTENT_CLASSNAME: string) => ReactNode'.
      Types of parameters 'option' and 'option' are incompatible.
        Type 'EuiComboBoxOptionOption<OptionData>' is not assignable to type 'ExpandedComboBoxOption'.
          Type 'EuiComboBoxOptionOption<OptionData>' is missing the following properties from type '{ _append: Element[]; _prepend: Element; serviceType: string; }': _append, _prepend, serviceType
  Overload 2 of 2, '(props: _EuiComboBoxProps<OptionData>, context: any): EuiComboBox<OptionData>', gave the following error.
    Type '(option: ExpandedComboBoxOption, searchValue: string, contentClassName: string) => React.JSX.Element' is not assignable to type '(option: EuiComboBoxOptionOption<OptionData>, searchValue: string, OPTION_CONTENT_CLASSNAME: string) => ReactNode'.ts(2769)
```

Consuming the render options:

```typescript
<EuiComboBox
   ...
   renderOption={renderOption}
   ...
/>
```

Reply:
OK @TattdCodeMonkey now I fiex the Type instad of casting as you suggested https://github.com/elastic/kibana/pull/199464/commits/f2b682a9f23321851f364fd345933144a7db0bf8